src/components/Detail.tsx

```diff
@@ -1,27 +1,20 @@
 import '../styles/Detail.scss';
 
-type DetailProps =
-  | {
-      title: string;
-      textContent?: string | number;
-      arrayOfContent: { name: string }[];
-    }
-  | {
-      title: string;
-      textContent: string | number;
-      arrayOfContent?: { name: string }[];
-    };
+type DetailProps = {
+  title: string;
+  textContent: string | number | undefined | null;
```

Comment:
if textContent takes 'undefined' and 'null', maybe this property is optional?

Reply:
I'm monitoring this property inside component

Reply:
textContent: string | number | undefined | null; >> textContent?: string | number | null;
It is easier to see that property is optional

Reply:
i would recommend you to change textContent variable to some other name, "textContent" suits better for String type.